File: 2022/day_03/part_1.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::collections::HashSet;

fn main() {
    let file = File::open("input.txt").unwrap();
    let lines = io::BufReader::new(file).lines();

    let mut priority_total = 0;

    for line_result in lines {
        let line = line_result.unwrap();
        let i_middle = line.len() / 2;

        let left = line[..i_middle].chars().collect::<HashSet<_>>();
        let right = line[i_middle..].chars().collect::<HashSet<_>>();

        let mut intersection = left.intersection(&right);

        let c = intersection.next().unwrap();
        let mut n = *c as i32;
        
        if n >= 97 {
            // Lowercase
            n = n - 97 + 1;
        } else {
            // Lowercase
            n = n - 65 + 27;
        }

        priority_total += n;
    }

    
    println!("{}", priority_total);
}

File: 2024/day_04/part_1.py
grid = []
with open('input.txt') as in_file:
    for line in in_file:
        line = line.strip()
        grid.append(list(line))

directions = [
    (1, 0),
    (1, 1),
    (0, 1),
    (-1, 1),
    (-1, 0),
    (-1, -1),
    (0, -1),
    (1, -1),
]

x = 1
y = 1
for (a, b) in directions:
    print(grid[y + b][x + a])
